examples/magicexample.html

```diff
@@ -0,0 +1,140 @@
+        }
+
+    </style>
+
+<script id="choropleth" type="text/cartocss">
+/** choropleth visualization */
+
+/** simple visualization */
+
+#ne_10m_populated_places_simple_6{
+  marker-fill-opacity: 0.8;
+  marker-line-color: #FFF;
+  marker-line-width: 1;
+  marker-line-opacity: 1;
+  marker-width: 10;
+  marker-fill: #FFFFB2;
+  marker-allow-overlap: true;
+}
+#ne_10m_populated_places_simple_6 [ pop_max <= 35676000] {
+   marker-fill: #B10026;
+}
+#ne_10m_populated_places_simple_6 [ pop_max <= 534555] {
+   marker-fill: #E31A1C;
+}
+#ne_10m_populated_places_simple_6 [ pop_max <= 211138] {
+   marker-fill: #FC4E2A;
+}
+#ne_10m_populated_places_simple_6 [ pop_max <= 109023] {
+   marker-fill: #FD8D3C;
+}
+#ne_10m_populated_places_simple_6 [ pop_max <= 59575] {
+   marker-fill: #FEB24C;
+}
+#ne_10m_populated_places_simple_6 [ pop_max <= 31253] {
+   marker-fill: #FED976;
+}
+#ne_10m_populated_places_simple_6 [ pop_max <= 11739] {
+   marker-fill: #FFFFB2;
+}
+
+</script>
+
+</head>
+<body>
+<div id="map">
+</div>
+<div id="info">
+</div>
+<div id="altstyle">
+<h2>CLICK HERE TO CHANGE STYLE</h1>
+</div>
+  <script src="../vendor/leaflet.js"></script>
+  <script src="../vendor/d3.v3.js"></script>
+  <script src="../dist/d3.cartodb.full.js"></script>
+  <script src="https://code.jquery.com/jquery-2.1.4.min.js"></script>
+  <script>
+  var c = [40.71512685201709, -74.00201797485352]
+  var map = new L.Map("map", {center: c, zoom: 3})
+
+  L.tileLayer('http://{s}.basemaps.cartocdn.com/light_all/{z}/{x}/{y}.png', {
+  attribution: '&copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors, &copy; <a href="http://cartodb.com/attributions">CartoDB</a>'
+  }).addTo(map);
+
+  var cartoDBd3Layer = new L.CartoDBd3Layer({
+    user: 'development',
+    table: 'ne_10m_populated_places_simple_6',
+    cartocss: document.getElementById('choropleth').innerHTML,
+    tiler_template: "http://development.localhost.lan:8181"
+  }).addTo(map);
+
+  var alternate = "/** choropleth visualization */ #ne_10m_populated_places_simple_6{ marker-fill-opacity: 0.8; marker-line-color: #FFF; marker-line-width: 1; marker-line-opacity: 1; marker-width: 10; marker-fill: #FFFFCC; marker-allow-overlap: true; } #ne_10m_populated_places_simple_6 [ pop_max <= 35676000] { marker-fill: #0C2C84; } #ne_10m_populated_places_simple_6 [ pop_max <= 534555] { marker-fill: #225EA8; } #ne_10m_populated_places_simple_6 [ pop_max <= 211138] { marker-fill: #1D91C0; } #ne_10m_populated_places_simple_6 [ pop_max <= 109023] { marker-fill: #41B6C4; } #ne_10m_populated_places_simple_6 [ pop_max <= 59575] { marker-fill: #7FCDBB; } #ne_10m_populated_places_simple_6 [ pop_max <= 31253] { marker-fill: #C7E9B4; } #ne_10m_populated_places_simple_6 [ pop_max <= 11739] { marker-fill: #FFFFCC; }";
+
+  cartoDBd3Layer.on(0, "featureOut", function(f, d3selection){
+    d3selection.reset();
+  });
+
+  cartoDBd3Layer.on(0, "featureOver", function(f, d3selection){
```

Comment:
@alonsogarciapablo yeah but note that cartoDBd3Layer can contain more than one layer, hence the index